lib/ruby_lsp/executor.rb

```diff
@@ -236,11 +238,33 @@ def inlay_hint(uri, range)
       ).returns(T.nilable(T::Array[Interface::CodeAction]))
     end
     def code_action(uri, range, context)
-      start_line = range.dig(:start, :line)
-      end_line = range.dig(:end, :line)
       document = @store.get(uri)
 
-      Requests::CodeActions.new(uri, document, start_line..end_line, context).run
+      Requests::CodeActions.new(uri, document, range, context).run
+    end
+
+    sig do
+      params(params: T::Hash[Symbol, T.untyped])
+        .returns(T.any(Interface::CodeAction, Requests::CodeActionResolve::Error))
```

Comment:
I think this method only returns an `Interface::CodeAction`. The raised error is not considered a return of the method.